packages/motherduck/test/Adaptor.test.js

```diff
@@ -0,0 +1,885 @@
+      );
+      const result = await operation(baseState);
+      
+      expect(result.data).to.deep.equal(user);
+      expect(result.response).to.exist;
+      expect(result.response.command).to.equal('INSERT');
+    });
+
+    it('should handle special characters in string values', async () => {
+      const userWithQuotes = {
+        name: "O'Malley",
+        email: "test@example.com",
+        bio: 'Says "Hello World"'
+      };
+      
+      const insertOp = execute(insert('users', userWithQuotes));
+      const result = await insertOp(baseState);
+      
+      // Should handle special characters without errors
+      expect(result.data).to.deep.equal(userWithQuotes);
+      expect(result.response.command).to.equal('INSERT');
+    });
+
+    it('should handle null and undefined values', async () => {
+      const userWithNulls = {
+        name: 'John Doe',
+        age: null,
+        email: undefined,
+        active: true
+      };
+      
+      const insertOp = execute(insert('users', userWithNulls));
+      const result = await insertOp(baseState);
+      
+      expect(result.data).to.exist;
+      expect(result.response.command).to.equal('INSERT');
+    });
+
+    it('should expand references in table name and data', async () => {
+      const userData = { name: 'Dynamic User', age: 25 };
+      
+      const dynamicInsert = execute(insert('users', userData));
+      
+      const result = await dynamicInsert(baseState);
+      expect(result.data).to.deep.equal(userData);
+      expect(result.response.command).to.equal('INSERT');
+    });
+
+    it('should validate table name for SQL injection', async () => {
+      const maliciousTableName = "users; DROP TABLE users; --";
+      const insertOp = execute(insert(maliciousTableName, { name: 'Test' }));
+      
+      try {
+        await insertOp(baseState);
+        expect.fail('Should have thrown validation error');
+      } catch (error) {
+        expect(error.message).to.include('Invalid SQL identifier');
+      }
+    });
+  });
+
+  describe('insertMany', () => {
+    it('should insert multiple user records in batch', async () => {
+      const users = fixtures.users.slice(0, 2);
+      const insertOp = execute(insertMany('users', users));
+      const result = await insertOp(baseState);
+      
+      expect(result.data).to.deep.equal(users);
+      expect(result.response).to.exist;
+      expect(result.response.rowCount).to.equal(users.length);
+    });
+
+    it('should handle large batch inserts efficiently', async () => {
+      // Generate 50 test records
+      const largeDataset = Array.from({ length: 50 }, (_, i) => ({
+        id: i + 1,
+        name: `User ${i + 1}`,
+        age: 20 + (i % 30),
+        email: `user${i + 1}@example.com`
+      }));
+      
+      const insertOp = execute(insertMany('users', largeDataset));
+      const result = await insertOp(baseState);
+      
+      // Should handle large batch efficiently
```

Comment:
This is a good test but why is this "efficient"? :thinking:  What is actually under test here?